app/experimenter/nimbus-ui/src/hooks/useSearchParamsState.tsx

```diff
@@ -42,7 +42,7 @@ export function useSearchParamsState(storageKey?: string) {
     } else {
       const savedParams = storage.current[storageKey];
       if (savedParams) {
-        navigate(`${location.pathname}?${savedParams}`);
+        navigate(`${location.pathname}?${savedParams}`, { replace: true });
```

Comment:
This is the actual fix, just a one-liner. The rest of this is mainly test enhancements to assert exactly when navigate gets called with `{ replace: true }`